app/components/task/details/task-dependency-browser/task-dependency-browser.component.ts

```diff
@@ -30,12 +30,18 @@ export class TaskDependencyBrowserComponent implements OnChanges {
             this._tasks.clear();
         }
 
-        if (changes.jobId || ComponentUtils.recordChangedId(changes.task)) {
+        // Task initially loaded with no dedendsOn. Change to task properties was not handled here.
+        if (changes.jobId || ComponentUtils.recordChangedId(changes.task) || this._hasUnloadedDependencies) {
             this._loaded = 0;
```

Comment:
This only gets called twice now if it needs to be. The first time and the next time if there was a change in the DP property